src/core/MOM_barotropic.F90

```diff
@@ -1052,6 +1056,30 @@ subroutine btstep(U_in, V_in, eta_in, dt, bc_accel_u, bc_accel_v, forces, pbce,
     wt_v(i,J,k) = CS%frhatv(i,J,k) * visc_rem
   enddo ; enddo ; enddo
 
+  if (CS%wt_uv_fix) then
+    do j=js,je ; do I=is-1,ie ; Iwt_u_tot(I,j) = wt_u(I,j,1) ; enddo ; enddo
+    do k=2,nz ; do j=js,je ; do I=is-1,ie
+      Iwt_u_tot(I,j) = Iwt_u_tot(I,j) + wt_u(I,j,k)
+    enddo ; enddo ; enddo
+    do j=js,je ; do I=is-1,ie
+      Iwt_u_tot(I,j) = G%mask2dCu(I,j) / (Iwt_u_tot(I,j) + h_neglect)
```

Comment:
The denominator of this expression is dimensionally inconsistent, as is the expression on line 1076.  `Iwt_u_tot` is nondimensional, whereas `h_neglect` has units of [H ~> m or kg m-2].  This is essentially an Adcroft reciprocal, which could be implemented here without adding another value as `if (abs(Iwt_u_tot(I,j)) > 0.0) Iwt_u_tot(I,j) = G%mask2dCu(I,j) / Iwt_u_tot(I,j)` .